### PR TITLE
Added support for consecutive asterisks at the beginning or end

### PR DESCRIPTION
**Pull request by martinssipenko**
According to [CODEOWNERS Syntax doc](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax), CODEOWNERS file uses a pattern that follows most of the same rules used in [gitignore](https://git-scm.com/docs/gitignore#_pattern_format) files.

This Pull Requests adds support for consecutive asterisks at the beginning or end (`**/foo`, `foo/**`). The case where asterisks are in the middle (`a/**/b`) was already supported.

### CODE REVIEW

**Comment by timoschinkel** [comment]
Hi Martins,

Thank you for your pull request. I will try to have a closer look later this week. On first glance it looks good, although I'm wondering about the replacement of `/**` with `.*`, maybe that should be `\/.*`.

I see that something has changed in the configuration of Psalm causing the inspections to fail. I will look at that as well.

**Comment by martinssipenko** [comment]
I fixed Psalm in #21 and rebased this PR.

I just tried using `\/.*` instead of `.*` locally and the tests were failing. Honestly, I have not invested time to fully understand the inner working of the library, so I don't know what would be the best solution. I encountered an issue with `**/foo` case, so I added a test and patched code to make it work, duplicating the `**` case (`.*`) seemed the simplest option.

**Comment by timoschinkel** [comment]
No problem. I will try to look at your pull request before the end of the week. This pull request requires a bit more attention compared to #21.

**Comment by martinssipenko** [comment]
Sure, in the mean time I'll use my forked version. Thanks!

File: src/PatternMatcher.php [hunk 1]
```diff
@@ -62,6 +62,8 @@ private function isMatch(Pattern $pattern, string $filename): bool
                 '*' => '[^\/]+',
                 '?' => '[^\/]',
                 '**' => '.*',
+                '/**' => '.*',
```
Comment (timoschinkel) [inline src/PatternMatcher.php:65]:
```suggestion
                '/**' => '\/.*',
```

I think we do need to add the slashes. I have created a pull request on a test repository to verify some edge cases. I have added a coworker as codeowner to that repository: https://github.com/timoschinkel/test-repository/pull/6/files

For the rule `docs/**` that does apply to `docs/file.md` and `docs/folder/file.md`, but not to `docs.md`. 

These situations can be tested by adding the following lines to `\CodeOwners\Tests\PatternMatcherTest::provideNoMatchFoundExceptionIsThrownForFilename()`:

```php
            // **
            [new Pattern('**/foo', ['@owner']), 'foo.ext'],
            [new Pattern('foo/**', ['@owner']), 'foo.ext'],
```

**Review by timoschinkel — CHANGES_REQUESTED** [review]
Can you add the suggested test cases to `\CodeOwners\Tests\PatternMatcherTest::provideNoMatchFoundExceptionIsThrownForFilename()`? I think your current solution is too forgiving and will result in incorrect results.

**Comment by martinssipenko** [comment]
@timoschinkel Updated code as per your suggestions.

**Review by timoschinkel — CHANGES_REQUESTED** [review]
Sorry, but I did find an issue with the `**/` pattern. I've supplied a test case and I have verified this with a pull request on GitHub: https://github.com/timoschinkel/test-repository/pull/6/files

If we apply this fix then this change can be merged I think.

**Comment by martinssipenko** [comment]
@timoschinkel done

**Comment by timoschinkel** [comment]
Thank you for the contribution. I just released this in [2.1.0](https://github.com/timoschinkel/codeowners/releases/tag/2.1.0).
